dakara_server/library/management/commands/createworks.py

```diff
@@ -0,0 +1,225 @@
+
+        if not isinstance(dict_work, dict):
+            logger.warning((
+                "Value associated to key work {} "
+                "should be a dictionnary".format(work_title)))
+            return False
+
+        has_correct_struct = True
+        for field in dict_work:
+            if field not in field_names:
+                has_correct_struct = False
+                if work_title:
+                    logger.warning(
+                        "Incorrect field '{}' for '{}'.".format(
+                            field,
+                            work_title))
+                else:
+                    logger.warning(
+                        "Incorrect field '{}'".format(field))
+
+        return has_correct_struct
+
+    def creatework(self, work_type_entry, work_title, dict_work):
+        """Create or update a work in database."""
+        work_subtitles = dict_work['subtitles'] if 'subtitles' in dict_work else [""] # noqa E501
+
+        # get or create works
+        for work_subtitle in work_subtitles:
+            work_entry, work_created = Work.objects.get_or_create(
+                title__iexact=work_title,
+                work_type__id=work_type_entry.id,
+                subtitle__iexact=work_subtitle,
+                defaults={
+                    'title': work_title,
+                    'work_type': work_type_entry,
+                    'subtitle': work_subtitle}
+                )
+
+            if work_created:
+                logger.debug("Created work '{}' subtitled '{}'.".format(
+                    work_title,
+                    work_subtitle))
+
+            # get work alternative titles or create them
+            if 'alternative_titles' in dict_work:
+                for alt_title in dict_work['alternative_titles']:
+                    self.create_alternative_title(
+                            work_entry,
+                            alt_title)
+
+            # save work in the database
+            work_entry.save()
+
+    def create_alternative_title(self, work_entry, alt_title):
+        """Create work alternative title in the database if necessary."""
+        _, work_alt_title_created = WorkAlternativeTitle.objects.get_or_create(
+                title__iexact=alt_title,
+                work__title__iexact=work_entry.title,
+                work__subtitle__iexact=work_entry.subtitle,
+                defaults={
+                    'title': alt_title,
+                    'work': work_entry}
+                )
+
+        if work_alt_title_created:
+            logger.debug("Created alternative titles '{}' for '{}'.".format(
+                alt_title, work_entry.title))
+
+    def createworks(self):
+        """Create or update works provided."""
+        # parse the work file to get the data structure
+        try:
+            works = self.parser.parse_work(self.work_file)
+        except BaseException as exc:
+            raise CommandError("{}".format(str(exc))) from exc
```

Comment:
You can extract the message of the error just with: `str(exc)`. By the way, it would be nice to add an error prefix message, like: `CommandError("Error when reading the works file: {}".format(exc))`. `.format` stringifies its arguments by itself, iirc.